### PR TITLE
Temporary adding Apache Lucene repositories for snapshots

### PR DESCRIPTION
**Pull request by reta**
Signed-off-by: Andriy Redko <andriy.redko@aiven.io>

### Description
Temporary adding Apache Lucene repositories for snapshots
 
### Issues Resolved
Fixes build issues (after https://github.com/opensearch-project/OpenSearch/pull/2924)
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: gradle/code-coverage.gradle [hunk 1]
```diff
@@ -11,6 +11,13 @@ apply plugin: 'jacoco'
 repositories {
   mavenCentral()
   gradlePluginPortal()
+  // TODO: Find the way to use the repositories from RepositoriesSetupPlugin
```
Comment (reta) [inline gradle/code-coverage.gradle:14]: @nknize @dblock I don't have better solution at the moment but this fix unblocks the builds

**Review by nknize — APPROVED** [review]
👍 I have no issue with this for now.  Thanks for looking!

**Comment by reta** [comment]
> x   Gradle Check failure [6a875fe](https://github.com/opensearch-project/OpenSearch/commit/6a875fe0b282b6e5fb1785e4b154a58ebe85dc15) [Log 4714](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_4714.log)
> 
> [Reports 4714](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_4714_reports.zip)

For the record, `org.apache.lucene:lucene-analysis-smartcn:9.2.0-snapshot-f4f1f70` is not considered to be a 'snapshot' by Maven (it does not end with `snapshot`)

**Comment by reta** [comment]
> x   Gradle Check failure [dfd54c6](https://github.com/opensearch-project/OpenSearch/commit/dfd54c6e9c7b17c3e3fd071978cc2f66172a177c) [Log 4715](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_4715.log)
> 
> [Reports 4715](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_4715_reports.zip)

Hm ....

Tests with failures:
 - org.opensearch.snapshots.SharedClusterSnapshotRestoreIT.testDataFileCorruptionDuringRestore
 - org.opensearch.recovery.TruncatedRecoveryIT.testCancelRecoveryAndResume
 - org.opensearch.search.query.SearchQueryIT.testAllDocsQueryString
 - org.opensearch.snapshots.DedicatedClusterSnapshotRestoreIT.testRestoreShrinkIndex

**Comment by reta** [comment]
start gradle check

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
start gradle check-

**Comment by dblock** [comment]
@peterzhuamazon could you please check that @reta can do gradle check starts too? I think it's a separate permissions list

**Comment by reta** [comment]
> @peterzhuamazon could you please check that @reta can do gradle check starts too? I think it's a separate permissions list

I can, thanks for checking @dblock !

**Comment by reta** [comment]
> x   Gradle Check failure [ef05839](https://github.com/opensearch-project/OpenSearch/commit/ef058399332ff8b4873d845f1105ab5aff7fea3f) [Log 4719](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_4719.log)
> 
> [Reports 4719](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_4719_reports.zip)

It does not look good, seem like we have introduce instability somewhere ...

```
Tests with failures:
 - org.opensearch.action.search.TransportSearchIT.testCircuitBreakerReduceFail
 - org.opensearch.search.aggregations.bucket.LongTermsIT.testSingleValuedFieldOrderedBySingleValueSubAggregationAscAndTermsAsc
 - org.opensearch.search.aggregations.bucket.LongTermsIT.testSingleValuedFieldOrderedBySubAggregationAscMultiHierarchyLevels
 - org.opensearch.search.aggregations.metrics.HDRPercentilesIT.testOrderBySubAggregation
 - org.opensearch.indexing.IndexActionIT.testAutoGenerateIdNoDuplicates
 - org.opensearch.indices.IndicesRequestCacheIT.testCacheAggs
 - org.opensearch.snapshots.CloneSnapshotIT.testShardClone
 - org.opensearch.search.functionscore.FunctionScoreFieldValueIT.testFieldValueFactor
 - org.opensearch.search.basic.TransportSearchFailuresIT.testFailedSearchWithWrongQuery
 - org.opensearch.recovery.TruncatedRecoveryIT.testCancelRecoveryAndResume
 - org.opensearch.search.functionscore.FunctionScoreFieldValueIT.testFieldValueFactorExplain

2616 tests completed, 11 failed, 7 skipped
```

**Comment by reta** [comment]
start gradle check

**Comment by reta** [comment]
> x   Gradle Check failure [ef05839](https://github.com/opensearch-project/OpenSearch/commit/ef058399332ff8b4873d845f1105ab5aff7fea3f) [Log 4720](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_4720.log)
> 
> [Reports 4720](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_4720_reports.zip)

Uh ... 


```
Tests with failures:
 - org.opensearch.search.aggregations.bucket.terms.StringTermsIT.testSingleValuedFieldOrderedBySingleBucketSubAggregationAsc
 - org.opensearch.index.FinalPipelineIT.testFinalPipelineOfNewDestinationIsInvoked
 - org.opensearch.search.aggregations.bucket.LongTermsIT.testOtherDocCount
 - org.opensearch.search.aggregations.bucket.LongTermsIT.testMultiValuedFieldWithValueScript
 - org.opensearch.search.aggregations.bucket.LongTermsIT.testMultiValuedFieldWithValueScriptNotUnique
 - org.opensearch.search.aggregations.bucket.LongTermsIT.testScriptCaching
 - org.opensearch.search.aggregations.bucket.LongTermsIT.testPartiallyUnmapped
 - org.opensearch.search.aggregations.bucket.LongTermsIT.testSingleValuedFieldOrderedBySingleValueSubAggregationAscWithTermsSubAgg
 - org.opensearch.search.aggregations.bucket.LongTermsIT.testSingleValueFieldWithPartitionedFiltering
 - org.opensearch.search.aggregations.bucket.LongTermsIT.testSingleValuedFieldOrderedBySingleBucketSubAggregationAsc
 - org.opensearch.search.aggregations.bucket.LongTermsIT.testSingleValuedFieldOrderedBySubAggregationAscMultiHierarchyLevels
 - org.opensearch.search.aggregations.bucket.LongTermsIT.testSingleValuedFieldOrderedByMultiValueSubAggregationAsc
 - org.opensearch.search.aggregations.bucket.LongTermsIT.testSingleValuedFieldOrderedByMultiValueSubAggregationDesc
 - org.opensearch.search.aggregations.bucket.LongTermsIT.testSingleValuedFieldOrderedByMultiValueExtendedStatsAsc
 - org.opensearch.search.aggregations.bucket.LongTermsIT.testPartiallyUnmappedWithFormat
 - org.opensearch.search.aggregations.bucket.LongTermsIT.testSingleValuedFieldWithValueScript
 - org.opensearch.search.fields.SearchFieldsIT.testStoredFields
 - org.opensearch.routing.AliasRoutingIT.testAliasSearchRouting
 - org.opensearch.search.aggregations.bucket.LongTermsIT.classMethod
 - org.opensearch.search.aggregations.bucket.MinDocCountIT.testLongScriptCountAsc
 - org.opensearch.search.aggregations.bucket.MinDocCountIT.testDoubleTermAsc
 - org.opensearch.search.aggregations.bucket.MinDocCountIT.testStringCountAscWithInclude
 - org.opensearch.search.profile.query.QueryProfilerIT.testProfileQuery
 - org.opensearch.search.profile.query.QueryProfilerIT.testPhrase
 - org.opensearch.search.profile.query.QueryProfilerIT.testSimpleMatch
 - org.opensearch.search.profile.query.QueryProfilerIT.testDisMaxRange
 - org.opensearch.search.profile.query.QueryProfilerIT.testNoProfile
 - org.opensearch.search.profile.query.QueryProfilerIT.testCollapsingBool
```

```
2> REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.opensearch.search.profile.query.QueryProfilerIT.testProfileQuery" -Dtests.seed=2E6E80CA89F5582E -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=et -Dtests.timezone=Asia/Harbin -Druntime.java=17
  2> MasterNotDiscoveredException[null]
        at app//org.opensearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction$2.onTimeout(TransportMasterNodeAction.java:275)
        at app//org.opensearch.cluster.ClusterStateObserver$ContextPreservingListener.onTimeout(ClusterStateObserver.java:369)
        at app//org.opensearch.cluster.ClusterStateObserver$ObserverClusterStateListener.onTimeout(ClusterStateObserver.java:287)
        at app//org.opensearch.cluster.service.ClusterApplierService$NotifyTimeout.run(ClusterApplierService.java:692)
        at app//org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:741)
        at java.base@17.0.2/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base@17.0.2/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base@17.0.2/java.lang.Thread.run(Thread.java:833)

    com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=1030, name=opensearch[node_s1][transport_worker][T#2], state=RUNNABLE, group=TGRP-QueryProfilerIT]

        Caused by:
        java.lang.AssertionError: No thread should be marked active when task finishes
            at __randomizedtesting.SeedInfo.seed([2E6E80CA89F5582E]:0)
            at org.opensearch.tasks.TaskResourceTrackingService.stopTracking(TaskResourceTrackingService.java:117)
```

**Comment by reta** [comment]
@dblock @nknize I cannot get green build :( thinking about rolling back https://github.com/opensearch-project/OpenSearch/pull/2639

**Comment by reta** [comment]
start gradle check

**Comment by reta** [comment]
start gradle check

**Comment by dblock** [comment]
Are you saying we introduced this break in https://github.com/opensearch-project/OpenSearch/pull/2639? Paging @tushar-kharbanda72, let's see if we can fix this before we rollback?

**Comment by reta** [comment]
> Are you saying we introduced this break in #2639? Paging @tushar-kharbanda72, let's see if we can fix this before we rollback?

It looks like it, many exceptions in logs (java.lang.AssertionError: No thread should be marked active when task finishes):

```

    com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=1030, name=opensearch[node_s1][transport_worker][T#2], state=RUNNABLE, group=TGRP-QueryProfilerIT]

        Caused by:
        java.lang.AssertionError: No thread should be marked active when task finishes
            at __randomizedtesting.SeedInfo.seed([2E6E80CA89F5582E]:0)
            at org.opensearch.tasks.TaskResourceTrackingService.stopTracking(TaskResourceTrackingService.java:117)
```

**Comment by reta** [comment]
start gradle check

**Comment by dblock** [comment]
I am ok reverting that PR if it helps @reta.

**Comment by reta** [comment]
@dblock we may deadlock here - we need this guy to unblock the Lucene snapshots issue, and https://github.com/opensearch-project/OpenSearch/pull/3046 may fail because of that

**Comment by nknize** [comment]
Just catching up on this.... I see a gradle success w/ no commits or rebases since the last massive failure. Care to summarize?

**Comment by reta** [comment]
> Just catching up on this.... I see a gradle success w/ no commits or rebases since the last massive failure. Care to summarize?

Yeah, the builds are super unstable, in report, a lot of exceptions like https://github.com/opensearch-project/OpenSearch/pull/3042#issuecomment-1106743308 pop up

**Comment by nknize** [comment]
> Yeah, the builds are super unstable, in report

What do you mean by "in report"? Do you mean this as in the general "build report" or are you referring to the added code coverage report?

**Comment by reta** [comment]
> > Yeah, the builds are super unstable, in report
> 
> What do you mean by "in report"? Do you mean this as in the general "build report" or are you referring to the added code coverage report?

Oops, sorry, in build logs and build reports

**Pull request merged**